libraries/disp3D/engine/model/items/sensordata/cpusensordatatreeitem.cpp

```diff
@@ -0,0 +1,372 @@
+
+CpuSensorDataTreeItem::~CpuSensorDataTreeItem()
+{
+    if(m_pSensorRtDataWorker->isRunning()) {
+        m_pSensorRtDataWorker->stop();
+        delete m_pSensorRtDataWorker;
+    }
+}
+
+
+//*************************************************************************************************************
+
+void CpuSensorDataTreeItem::init(const MatrixX3f& matSurfaceVertColor,
+                              const MNEBemSurface& bemSurface,
+                              const FiffInfo& fiffInfo,
+                              const QString& sSensorType,
+                              const double dCancelDist,
+                              const QString& sInterpolationFunction)
+{
+    this->setData(0, Data3DTreeModelItemRoles::RTData);
+
+    if(!m_pSensorRtDataWorker) {
+        m_pSensorRtDataWorker = new RtSensorDataWorker();
+    }
+
+    connect(m_pSensorRtDataWorker.data(), &RtSensorDataWorker::newRtData,
+            this, &CpuSensorDataTreeItem::onNewRtData);
+
+    // map passed sensor type string to fiff constant
+    fiff_int_t sensorTypeFiffConstant;
+    if (sSensorType.toStdString() == std::string("MEG")) {
+        sensorTypeFiffConstant = FIFFV_MEG_CH;
+    } else if (sSensorType.toStdString() == std::string("EEG")) {
+        sensorTypeFiffConstant = FIFFV_EEG_CH;
+    } else {
+        qDebug() << "SensorDataTreeItem::init - unknown sensor type. Returning ...";
+        return;
+    }
+
+    //fill QVector with the right sensor positions
+    QVector<Vector3f> vecSensorPos;
+    m_iUsedSensors.clear();
+    int iCounter = 0;
+    for(const FiffChInfo &info : fiffInfo.chs) {
+        //Only take EEG with V as unit or MEG magnetometers with T as unit
+        if(info.kind == sensorTypeFiffConstant && (info.unit == FIFF_UNIT_T || info.unit == FIFF_UNIT_V)) {
+            vecSensorPos.push_back(info.chpos.r0);
+
+            //save the number of the sensor
+            m_iUsedSensors.push_back(iCounter);
+        }
+        iCounter++;
+    }
+
+    //Create bad channel idx list
+    for(const QString &bad : fiffInfo.bads) {
+        m_iSensorsBad.push_back(fiffInfo.ch_names.indexOf(bad));
+    }
+
+    setCancelDistance(dCancelDist);
+    setInterpolationFunction(sInterpolationFunction);
+
+    m_pSensorRtDataWorker->calculateSurfaceData(bemSurface,
+                                                vecSensorPos,
+                                                fiffInfo,
+                                                sensorTypeFiffConstant);
+
+    m_pSensorRtDataWorker->setSurfaceColor(matSurfaceVertColor);
+
+    m_bIsDataInit = true;
+}
+
+
+//*************************************************************************************************************
+
+void CpuSensorDataTreeItem::addData(const MatrixXd &tSensorData)
+{
+    if(!m_bIsDataInit) {
+        qDebug() << "SensorDataTreeItem::addData - sensor data item has not been initialized yet!";
```

Comment:
use correct clas names in debug warnings etc "CpuSensorDataTreeItem::addData ...